### PR TITLE
Add BVTs for complianceConfig

### PR DESCRIPTION
**Pull request by JeffeyL**
This commit tests the ability of complianceConfig to configure compliance in specific situations.

Ref: https://issues.redhat.com/browse/ACM-11661

### CODE REVIEW

File: test/integration/operator_policy_complianceconfig_test.go
Comment (mprahl) [inline test/integration/operator_policy_complianceconfig_test.go:14]:
```suggestion
var _ = Describe("GRC: [P1][Sev1][policy-grc] Test OperatorPolicy complianceConfig",
```

File: test/integration/operator_policy_complianceconfig_test.go
Comment (mprahl) [inline test/integration/operator_policy_complianceconfig_test.go:15]: I'd rather this be an SVT test since it's a more advanced feature.
```suggestion
	Ordered, func() {
```

File: test/integration/operator_policy_complianceconfig_test.go
Comment (mprahl) [inline test/integration/operator_policy_complianceconfig_test.go:33]: OperatorPolicy will automatically create the namespace if it's missing, so I think this and the other namespace creation command can be removed.

File: test/integration/operator_policy_complianceconfig_test.go
Comment (mprahl) [inline test/integration/operator_policy_complianceconfig_test.go:113]: This `It` should be combined with the one above it since they are part of the same test.

File: test/integration/operator_policy_complianceconfig_test.go
Comment (mprahl) [inline test/integration/operator_policy_complianceconfig_test.go:140]: This `It` should be combined with the one above it since they are part of the same test.

File: test/integration/operator_policy_complianceconfig_test.go
Comment (mprahl) [inline test/integration/operator_policy_complianceconfig_test.go:191]: This `It` should be combined with the one above it since they are part of the same test.

File: test/integration/operator_policy_complianceconfig_test.go
Comment (mprahl) [inline test/integration/operator_policy_complianceconfig_test.go:125]: Could you use a regex instead of hardcoding the version which will likely change?

File: test/integration/operator_policy_complianceconfig_test.go
Comment (mprahl) [reply test/integration/operator_policy_complianceconfig_test.go:140]: Performing the upgrade doesn't seem to be testing `complianceConfig`. I would consider this an unrelated test and remove it. This is covered elsewhere.

File: test/integration/operator_policy_complianceconfig_test.go
Comment (mprahl) [inline test/integration/operator_policy_complianceconfig_test.go:155]: Since it's for a different testcase, could this be in a separate `Describe`?

File: test/integration/operator_policy_complianceconfig_test.go
Comment (mprahl) [inline test/integration/operator_policy_complianceconfig_test.go:14]:
```suggestion
var _ = Describe("RHACM4K-48381 GRC: [P1][Sev1][policy-grc] Test OperatorPolicy complianceConfig",
```

Comment (mprahl) [reply test/integration/operator_policy_complianceconfig_test.go:14]: This will allow QE to track the test result with a Polarion ID

File: test/resources/operator_policy/test-op-complianceconfig-1.yaml
Comment (yiraeChristineKim) [inline test/resources/operator_policy/test-op-complianceconfig-1.yaml:52]: Could you change to `placement`

File: test/resources/operator_policy/test-op-complianceconfig-1.yaml
Comment (yiraeChristineKim) [inline test/resources/operator_policy/test-op-complianceconfig-1.yaml:57]:
```suggestion
apiVersion: cluster.open-cluster-management.io/v1beta1
kind: Placement
metadata:
  name: test-op-complianceconfig-1-plr
spec:
   predicates:
  - requiredClusterSelector:
      labelSelector:
        matchExpressions: []
```

Comment (JeffeyL) [reply test/resources/operator_policy/test-op-complianceconfig-1.yaml:57]: Should I do this for the other files as well?

Comment (yiraeChristineKim) [reply test/resources/operator_policy/test-op-complianceconfig-1.yaml:57]: oh yes!! Good good

File: test/resources/operator_policy/test-op-complianceconfig-1.yaml
Comment (yiraeChristineKim) [inline test/resources/operator_policy/test-op-complianceconfig-1.yaml:45]:
```suggestion
  kind: Placement
  apiGroup: cluster.open-cluster-management.io
```

**Review by yiraeChristineKim — APPROVED** [review]
/hold for @mprahl

**Review by mprahl — APPROVED** [review]

**Comment by mprahl** [comment]
@JeffeyL you'll have to rebase after #846 is merged since I believe test-e2e-grc-framework will fail.

**Comment by dhaiducek** [comment]
@mprahl @JeffeyL What's the status here?

**Comment by mprahl** [comment]
@JeffeyL could you please rebase?

**Comment by JeffeyL** [comment]
Apologies, not sure how I missed this.

**Review by mprahl — APPROVED** [review]

**Comment by JeffeyL** [comment]
@mprahl What's the best way to figure out why `test-e2e-grc-framework` is failing? I tried looking at the logs but I couldn't deduce much.

**Comment by mprahl** [comment]
@JeffeyL this is unrelated to your PR, so I'll just go ahead and merge it. Let's keep a close eye on this test during the periodic test runs.

**Pull request merged**
